### DISCUSSION TITLE
bootstrap boot stuck on Skipped "user-ca-bundle-config.yaml"

### DISCUSSION THREAD

**Discussion by ricky-chaoju** [okd-project/okd#1888] [General]
bootstrap boot stuck on Skipped "user-ca-bundle-config.yaml" 

bootstrap stuck at ''Feb 15 15:23:10 bootstrap.okd.infinirc.com cluster-bootstrap[28164]: Skipped "user-ca-bundle-config.yaml" configmaps.v1./user-ca-bundle -n openshift-config as it already exists
Feb 15 15:23:10 bootstrap.okd.infinirc.com bootkube.sh[28141]: Skipped "user-ca-bundle-config.yaml" configmaps.v1./user-ca-bundle -n openshift-config as it already exists''.

The installation is stuck below
   -n openshift-kube-controller-manager-operator as it already exists
Feb 15 15:23:08 bootstrap.okd.infinirc.com bootkube.sh[28141]: Skipped "secret-csr-signer-signer.yaml" secrets.v1./csr-signer-signer -n openshift-kube-controller -manager-operator as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com cluster-bootstrap[28164]: Skipped "secret-initial-kube-controller-manager-service-account-private-key.yaml" secrets.v1./initial- service-account-private-key -n openshift-config as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com bootkube.sh[28141]: Skipped "secret-initial-kube-controller-manager-service-account-private-key.yaml" secrets.v1./initial- service-account-private-key -n openshift-config as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com cluster-bootstrap[28164]: Skipped "secret-kube-apiserver-to-kubelet-signer.yaml" secrets.v1./kube-apiserver-to-kubelet- signer -n openshift-kube-apiserver-operator as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com bootkube.sh[28141]: Skipped "secret-kube-apiserver-to-kubelet-signer.yaml" secrets.v1./kube-apiserver-to-kubelet- signer -n openshift-kube-apiserver-operator as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com cluster-bootstrap[28164]: Skipped "secret-loadbalancer-serving-signer.yaml" secrets.v1./loadbalancer-serving-signer -n openshift-kube-apiserver -operator as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com bootkube.sh[28141]: Skipped "secret-loadbalancer-serving-signer.yaml" secrets.v1./loadbalancer-serving-signer -n openshift-kube-apiserver -operator as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com cluster-bootstrap[28164]: Skipped "secret-localhost-serving-signer.yaml" secrets.v1./localhost-serving-signer -n openshift-kube-apiserver -operator as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com bootkube.sh[28141]: Skipped "secret-localhost-serving-signer.yaml" secrets.v1./localhost-serving-signer -n openshift-kube-apiserver -operator as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com cluster-bootstrap[28164]: Skipped "secret-service-network-serving-signer.yaml" secrets.v1./service-network-serving-signer -n openshift -kube-apiserver-operator as it already exists
Feb 15 15:23:09 bootstrap.okd.infinirc.com bootkube.sh[28141]: Skipped "secret-service-network-serving-signer.yaml" secrets.v1./service-network-serving-signer -n openshift -kube-apiserver-operator as it already exists
Feb 15 15:23:10 bootstrap.okd.infinirc.com cluster-bootstrap[28164]: Skipped "user-ca-bundle-config.yaml" configmaps.v1./user-ca-bundle -n openshift-config as it already exists
Feb 15 15:23:10 bootstrap.okd.infinirc.com bootkube.sh[28141]: Skipped "user-ca-bundle-config.yaml" configmaps.v1./user-ca-bundle -n openshift-config as it already exists

**Version**
OKD 4.15.0-0.okd-2024-02-10-035534
UPI mode
fedora-coreos-39.20240128.3.0-live.x86_64.iso
<img width="1242" alt="截圖 2024-02-15 晚上11 38 55" src="https://github.com/okd-project/okd/assets/109416633/701a4310-2377-4eb7-aac8-2fb1e1334b51">
<img width="852" alt="截圖 2024-02-15 晚上11 39 18" src="https://github.com/okd-project/okd/assets/109416633/5d47efbb-a2fc-4a13-b3e5-ef406265f936">

**Comment by vrutkovs** [comment]
Please attach [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)

**Reply by ricky-chaoju** [reply]
Hello, below is my bootstrap log. Please take a look. Thank you.
https://drive.google.com/file/d/1mvwty0LSsfrkeEVXkekVz4RWW52R_etF/view?usp=sharing

**Reply by vrutkovs** [reply]
This is not the entire log bundle archive it has generated, right?

**Reply by ricky-chaoju** [reply]
I use `journalctl -u bootkube.service > bootkube.log`
Is there a more recommended way to export logs? Thanks

**Reply by vrutkovs** [reply]
`./openshift-install gather bootstrap --dir` would include the required information. See [documentation](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)

**Reply by ricky-chaoju** [reply]
Hello, my 10.0.7.2 is bootstrap, and 10.0.7.3 is bootstrap that has not yet been booted. The following is the log of my installation of okd 4.10. It is the same as the installation of 4.15. Skipped "user-ca-bundle-config.yaml" configmaps.v1. /user-ca-bundle -n openshift-config as it already exists
Thanks
https://drive.google.com/drive/folders/1KGnxZbB04txi7ZMBiXNWZZ4cvtGkbme_?usp=drive_link

**Reply by vrutkovs** [reply]
>and 10.0.7.3 is bootstrap that has not yet been booted

you have two bootstrap nodes? Why? The other nodes should be 3 masters - and they need to boot to continue

**Reply by ricky-chaoju** [reply]
I only have one bootstrap node, so I have to start the master at the same time to continue. Is there no problem so far? Thank you.
Sorry, it’s not very clear when I use it for the first time.

**Reply by vrutkovs** [reply]
Right, masters and workers are safe to start with bootstrap node - they'll wait for ignition to be generated.

The rest looks correct to me

**Reply by ricky-chaoju** [reply]
Thank you very much, I will try again.

**Reply by ricky-chaoju** [reply]
Hello, I installed it successfully, but there seems to be a problem with authentication and console. I have added DNS and cannot access the backend. I don’t plan to use the worker node for the time being. Can you please help me take a look? Thanks

Similar to the following error, but cannot be solved
https://github.com/openshift/installer/issues/3960
https://access.redhat.com/solutions/7000864
<img width="1261" alt="截圖 2024-02-17 下午3 57 02" src="https://github.com/okd-project/okd/assets/109416633/28adaed3-5939-4a1e-b1ae-764b1d6fc421">
<img width="2531" alt="截圖 2024-02-17 下午4 22 05" src="https://github.com/okd-project/okd/assets/109416633/7d0cd204-4108-4390-88e5-a63bdb1fb76c">

<img width="2560" alt="截圖 2024-02-17 下午4 22 27" src="https://github.com/okd-project/okd/assets/109416633/f4523857-6103-4ca6-8092-60a730b7a9a1">
<img width="2560" alt="截圖 2024-02-17 下午4 22 33" src="https://github.com/okd-project/okd/assets/109416633/e8ab5f79-c768-45cd-b2c5-cb9293ac859e">

my log:
https://drive.google.com/drive/folders/12eNGL9WbBQA-6mXvabbTJjXL5VdyW_eU?usp=sharing

**Reply by vrutkovs** [reply]
What exactly is the error? The bootstrap has completed, but was not removed. The cluster won't continue until its still there

**Reply by ricky-chaoju** [reply]
After the installation is complete, the following error message appears. The console cannot be accessed. Sorry, I am a student and am not familiar with it. I have tried on the Internet and cannot solve it. I have deleted bootstrap. Thanks
<img width="2531" alt="截圖 2024-02-17 晚上7 57 29" src="https://github.com/okd-project/okd/assets/109416633/a99b44da-6b8b-4f1e-87de-36841ccea059">

**Reply by ricky-chaoju** [reply]
my install yaml
```
apiVersion: v1
baseDomain: infinirc.com
compute:
- hyperthreading: Enabled
  name: worker
  replicas: 0
controlPlane:
  hyperthreading: Enabled
  name: master
  replicas: 3
metadata:
  name: okd
networking:
  clusterNetwork:
  - cidr: 10.128.0.0/14
    hostPrefix: 23
  networkType: OVNKubernetes
  serviceNetwork:
  - 172.30.0.0/16
platform:
  none: {}
fips: false
pullSecret: 'privacy'
sshKey: 'privacy
additionalTrustBundle: |
  -----BEGIN CERTIFICATE-----
  privacy
  -----END CERTIFICATE-----


imageDigestSources:
  - mirrors:
    - registry.infinirc.com:8443/openshift/okd
    source: quay.io/openshift/okd
  - mirrors:
    - registry.infinirc.com:8443/openshift/okd
    source: quay.io/openshift/okd-content

```

**Reply by ricky-chaoju** [reply]
Hello, I succeeded. I reset haproxy and restarted all nodes successfully. Thank you very much for your help.

<img width="2531" alt="截圖 2024-02-18 上午8 50 50" src="https://github.com/okd-project/okd/assets/109416633/7ac39f7f-906b-4279-b7f1-8b22a770b03f">
